### PR TITLE
Add RSA and AES encryption/decryption capabilities to crypto stdlib

### PR DESCRIPTION
**Pull request by ayomawdb**
## Purpose
Ballerina `crypto` stdlib does not have functions to support encryption. This PR adds `AES` and `RSA` encryption capabilities to stdlib with common cipher modes and padding schemes. 

Resolves: #9054

### RSA Encryption

The sample given below shows how to use encryption function `encryptRsaEcb` and decryption function `decryptRsaEcb` to perform RSA encryption and decryption.

```ballerina
import ballerina/io;
import ballerina/crypto;
import ballerina/encoding;

public function main() returns error? {
     // Input value for cryto operations
     string input = "Hello Ballerina";
     byte[] inputArr = input.toByteArray("UTF-8");

     crypto:KeyStore keyStore = { path: "/home/ballerina/keystore.p12", password: "ballerina" };

     // Public key used for encryption
     crypto:PublicKey publicKey = check crypto:decodePublicKey(keyStore = keyStore, keyAlias = "ballerina");

     // Private key used for decryption
     crypto:PrivateKey privateKey = check crypto:decodePrivateKey(keyStore = keyStore, keyAlias = "ballerina",
                                                            keyPassword = "ballerina");


     // Encrypt and decrypt input value using RSA ECB PKCS1 padding.
     byte[] output = check crypto:encryptRsaEcb(inputArr, publicKey);
     output = check crypto:decryptRsaEcb(output, privateKey);
     io:println("RSA ECB PKCS1 decrypted value: " + encoding:byteArrayToString(output));

     // Encrypt and decrypt input value using RSA ECB OAEPwithSHA512andMGF1 padding.
     output = check crypto:encryptRsaEcb(inputArr, publicKey, padding = crypto:OAEPwithSHA512andMGF1);
     output = check crypto:decryptRsaEcb(output, privateKey, padding = crypto:OAEPwithSHA512andMGF1);
     io:println("RSA ECB OAEPwithSHA512andMGF1 decrypted value: " + encoding:byteArrayToString(output));
}
```

### AES Encryption

The sample given below shows how to use encryption function `encryptAesCbc` and decryption function `decryptAesCbc` to perform AES encryption and decryption.

```ballerina
import ballerina/io;
import ballerina/crypto;
import ballerina/encoding;
import ballerina/math;

public function main() returns error? {
     // Input value for cryto operations
     string input = "Hello Ballerina!";
     byte[] inputArr = input.toByteArray("UTF-8");

     // Randomly generate a 128 bit key
     byte[16] keyArr = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
     foreach var i in 0...15 {
        keyArr[i] = check byte.convert(math:randomInRange(0, 255));
     }

     // Randomly generate a 128 bit IV
     byte[16] ivArr = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
     foreach var i in 0...15 {
        ivArr[i] = check byte.convert(math:randomInRange(0, 255));
     }

     // Encrypt and decrypt input value using AES CBC PKCS5 padding
     byte[] output = check crypto:encryptAesCbc(inputArr, keyArr, ivArr);
     output = check crypto:decryptAesCbc(output, keyArr, ivArr);
     io:println("AES CBC PKCS5 decrypted value: " + encoding:byteArrayToString(output));

     // Encrypt and decrypt input value using AES CBC no padding
     output = check crypto:encryptAesCbc(inputArr, keyArr, ivArr, padding = crypto:NONE);
     output = check crypto:decryptAesCbc(output, keyArr, ivArr, padding = crypto:NONE);
     io:println("AES CBC no padding decrypted value: " + encoding:byteArrayToString(output));

     // Encrypt and decrypt input value using AES GCM PKCS5 padding
     output = check crypto:encryptAesGcm(inputArr, keyArr, ivArr);
     output = check crypto:decryptAesGcm(output, keyArr, ivArr);
     io:println("AES GCM PKCS5 decrypted value: " + encoding:byteArrayToString(output));

     // Encrypt and decrypt input value using AES GCM no padding
     output = check crypto:encryptAesGcm(inputArr, keyArr, ivArr, padding = crypto:NONE);
     output = check crypto:decryptAesGcm(output, keyArr, ivArr, padding = crypto:NONE);
     io:println("AES GCM no padding decrypted value: " + encoding:byteArrayToString(output));

     // Encrypt and decrypt input value using AES ECB PKCS5 padding
     output = check crypto:encryptAesEcb(inputArr, keyArr);
     output = check crypto:decryptAesEcb(output, keyArr);
     io:println("AES ECB PKCS5 decrypted value: " + encoding:byteArrayToString(output));

     // Encrypt and decrypt input value using AES ECB no padding
     output = check crypto:encryptAesEcb(inputArr, keyArr, padding = crypto:NONE);
     output = check crypto:decryptAesEcb(output, keyArr, padding = crypto:NONE);
     io:println("AES ECB no padding decrypted value: " + encoding:byteArrayToString(output));
}
```

### CODE REVIEW

File: examples/crypto/crypto.bal [hunk 1]
```diff
@@ -90,4 +91,63 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesCbc(inputArr, rsaKeyArr, ivArr, padding = crypto:NONE);
+     output = check crypto:decryptAesCbc(output, rsaKeyArr, ivArr, padding = crypto:NONE);
+     io:println("AES CBC no padding decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES GCM PKCS5 padding
+     output = check crypto:encryptAesGcm(inputArr, rsaKeyArr, ivArr);
+     output = check crypto:decryptAesGcm(output, rsaKeyArr, ivArr);
+     io:println("AES GCM PKCS5 decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES GCM no padding
```
Comment (Shan1024) [inline examples/crypto/crypto.bal]:
```suggestion
     // Encrypt and decrypt input value using AES GCM no padding.
```

File: examples/crypto/crypto.bal [hunk 2]
```diff
@@ -90,4 +91,63 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesGcm(inputArr, rsaKeyArr, ivArr, padding = crypto:NONE);
+     output = check crypto:decryptAesGcm(output, rsaKeyArr, ivArr, padding = crypto:NONE);
+     io:println("AES GCM no padding decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES ECB PKCS5 padding
```
Comment (Shan1024) [inline examples/crypto/crypto.bal]:
```suggestion
     // Encrypt and decrypt input value using AES ECB PKCS5 padding.
```

File: examples/crypto/crypto.bal [hunk 3]
```diff
@@ -90,4 +91,63 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesEcb(inputArr, rsaKeyArr);
+     output = check crypto:decryptAesEcb(output, rsaKeyArr);
+     io:println("AES ECB PKCS5 decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES ECB no padding
```
Comment (Shan1024) [inline examples/crypto/crypto.bal]:
```suggestion
     // Encrypt and decrypt input value using AES ECB no padding.
```

File: examples/crypto/crypto.bal [hunk 4]
```diff
@@ -90,4 +91,63 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesEcb(inputArr, rsaKeyArr, padding = crypto:NONE);
+     output = check crypto:decryptAesEcb(output, rsaKeyArr, padding = crypto:NONE);
+     io:println("AES ECB no padding decrypted value: " + encoding:byteArrayToString(output));
+
+     // Public key used for RSA encryption
```
Comment (Shan1024) [inline examples/crypto/crypto.bal]:
```suggestion
     // Public key used for RSA encryption.
```

File: examples/crypto/crypto.bal [hunk 5]
```diff
@@ -90,4 +91,63 @@ public function main() returns error? {
...
+     crypto:PublicKey rsaPublicKey = check crypto:decodePublicKey(keyStore = keyStore, keyAlias = "ballerina");
+
+     // Private key used for RSA decryption
```
Comment (Shan1024) [inline examples/crypto/crypto.bal]:
```suggestion
     // Private key used for RSA decryption.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 6]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
\ No newline at end of file
+```
+
+### RSA Encryption
+
+The sample given below shows how to use encryption function `encryptRsaEcb` and decryption function `decryptRsaEcb` to perform RSA encryption and decryption.
+
+
+```ballerina
+import ballerina/io;
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]: Better to reorder imports in alphabetical order.

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 7]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+import ballerina/crypto;
+import ballerina/encoding;
+
+public function main() returns error? {
+     // Input value for cryto operations
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Input value for cryto operations.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 8]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     string input = "Hello Ballerina";
+     byte[] inputArr = input.toByteArray("UTF-8");
+
+     crypto:KeyStore keyStore = { path: "/home/ballerina/keystore.p12", password: "ballerina" };
+
+     // Public key used for encryption
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Public key used for encryption.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 9]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     crypto:PublicKey publicKey = check crypto:decodePublicKey(keyStore = keyStore, keyAlias = "ballerina");
+
+     // Private key used for decryption
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Private key used for decryption.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 10]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+}
+```
+
+### AES Encryption
+
+The sample given below shows how to use encryption function `encryptAesCbc` and decryption function `decryptAesCbc` to perform AES encryption and decryption.
+
+
+```ballerina
+import ballerina/io;
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]: Better to reorder imports alphabetically.

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 11]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+import ballerina/crypto;
+import ballerina/encoding;
+import ballerina/math;
+
+public function main() returns error? {
+     // Input value for cryto operations
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Input value for cryto operations.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 12]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     string input = "Hello Ballerina!";
+     byte[] inputArr = input.toByteArray("UTF-8");
+
+     // Randomly generate a 128 bit key
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Randomly generate a 128 bit key.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 13]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     byte[16] keyArr = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+     foreach var i in 0...15 {
+        keyArr[i] = check byte.convert(math:randomInRange(0, 255));
+     }
+
+     // Randomly generate a 128 bit IV
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Randomly generate a 128 bit IV.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 14]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     byte[16] ivArr = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+     foreach var i in 0...15 {
+        ivArr[i] = check byte.convert(math:randomInRange(0, 255));
+     }
+
+     // Encrypt and decrypt input value using AES CBC PKCS5 padding
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Encrypt and decrypt input value using AES CBC PKCS5 padding.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 15]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     byte[] output = check crypto:encryptAesCbc(inputArr, keyArr, ivArr);
+     output = check crypto:decryptAesCbc(output, keyArr, ivArr);
+     io:println("AES CBC PKCS5 decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES CBC no padding
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Encrypt and decrypt input value using AES CBC no padding.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 16]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesCbc(inputArr, keyArr, ivArr, padding = crypto:NONE);
+     output = check crypto:decryptAesCbc(output, keyArr, ivArr, padding = crypto:NONE);
+     io:println("AES CBC no padding decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES GCM PKCS5 padding
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Encrypt and decrypt input value using AES GCM PKCS5 padding.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 17]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesGcm(inputArr, keyArr, ivArr);
+     output = check crypto:decryptAesGcm(output, keyArr, ivArr);
+     io:println("AES GCM PKCS5 decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES GCM no padding
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Encrypt and decrypt input value using AES GCM no padding.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 18]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesGcm(inputArr, keyArr, ivArr, padding = crypto:NONE);
+     output = check crypto:decryptAesGcm(output, keyArr, ivArr, padding = crypto:NONE);
+     io:println("AES GCM no padding decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES ECB PKCS5 padding
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Encrypt and decrypt input value using AES ECB PKCS5 padding.
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 19]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     output = check crypto:encryptAesEcb(inputArr, keyArr);
+     output = check crypto:decryptAesEcb(output, keyArr);
+     io:println("AES ECB PKCS5 decrypted value: " + encoding:byteArrayToString(output));
+
+     // Encrypt and decrypt input value using AES ECB no padding
```
Comment (Shan1024) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
     // Encrypt and decrypt input value using AES ECB no padding.
```

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/CryptoUtils.java [hunk 20]
```diff
@@ -123,4 +149,199 @@ public static BError createCryptoError(Context context, String errMsg) {
...
+     *
+     * @param context BRE context used to raise error messages
+     * @param algorithmMode algorithm mode
+     * @param iv initialization vector for CBC and GCM mode
+     * @param tagSize tag size for GCM mode
+     * @return algorithm parameter specification
+     */
+    private static AlgorithmParameterSpec buildParameterSpec(Context context, String algorithmMode, byte[] iv,
+                                                             int tagSize) {
+        if (algorithmMode.equals("GCM")) {
```
Comment (Shan1024) [inline stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/CryptoUtils.java]: We can use a switch statement here.

File: examples/crypto/crypto.bal [hunk 21]
```diff
@@ -1,11 +1,12 @@
 import ballerina/io;
 import ballerina/crypto;
 import ballerina/encoding;
+import ballerina/math;
```
Comment (NatashaWso2) [inline examples/crypto/crypto.bal]: Shall we arrange the imports in alphabetical order?

File: stdlib/crypto/src/main/ballerina/crypto/Module.md [hunk 22]
```diff
@@ -138,4 +138,101 @@ public function main() returns error? {
...
+     crypto:KeyStore keyStore = { path: "/home/ballerina/keystore.p12", password: "ballerina" };
+
+     // Public key used for encryption.
+     crypto:PublicKey publicKey = check crypto:decodePublicKey(keyStore = keyStore, keyAlias = "ballerina");
+
+     // Private key used for decryption.
+     crypto:PrivateKey privateKey = check crypto:decodePrivateKey(keyStore = keyStore, keyAlias = "ballerina",
+                                                            keyPassword = "ballerina");
+
+
```
Comment (NatashaWso2) [inline stdlib/crypto/src/main/ballerina/crypto/Module.md]: Shall we remove this extra line?

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 23]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
     crypto:PrivateKey pk = {algorithm:"RSA"};
     return crypto:signRsaMd5(input, pk);
 }
+
+function testEncryptAesEcb(byte[] input, byte[] key, crypto:AesPadding padding = "PKCS5")
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
function testEncryptAesEcb(byte[] input, byte[] key, crypto:AesPadding padding = "PKCS5") returns byte[]|error {
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 24]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+returns byte[]|error {
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 25]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+    return crypto:encryptAesEcb(input, key, padding = padding);
+}
+
+function testDecryptAesEcb(byte[] input, byte[] key, crypto:AesPadding padding = "PKCS5")
+returns byte[]|error {
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 26]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
     crypto:PrivateKey pk = {algorithm:"RSA"};
     return crypto:signRsaMd5(input, pk);
 }
+
+function testEncryptAesEcb(byte[] input, byte[] key, crypto:AesPadding padding = "PKCS5")
+returns byte[]|error {
+    return crypto:encryptAesEcb(input, key, padding = padding);
+}
+
+function testDecryptAesEcb(byte[] input, byte[] key, crypto:AesPadding padding = "PKCS5")
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
function testDecryptAesEcb(byte[] input, byte[] key, crypto:AesPadding padding = "PKCS5") returns byte[]|error {
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 27]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+returns byte[]|error {
+    return crypto:decryptAesEcb(input, key, padding = padding);
+}
+
+function testEncryptAesCbc(byte[] input, byte[] key, byte[] iv, crypto:AesPadding padding = "PKCS5")
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
function testEncryptAesCbc(byte[] input, byte[] key, byte[] iv, crypto:AesPadding padding = "PKCS5") returns byte[]|error {
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 28]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+returns byte[]|error {
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 29]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+    return crypto:encryptAesCbc(input, key, iv, padding = padding);
+}
+
+function testDecryptAesCbc(byte[] input, byte[] key, byte[] iv, crypto:AesPadding padding = "PKCS5")
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
function testDecryptAesCbc(byte[] input, byte[] key, byte[] iv, crypto:AesPadding padding = "PKCS5") returns byte[]|error {
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 30]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+returns byte[]|error {
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 31]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+string path, string keyStorePassword, string keyAlias) returns byte[]|error {
+    crypto:KeyStore keyStore = {
+        path: path,
+        password: keyStorePassword
+    };
+    crypto:PublicKey pk = check crypto:decodePublicKey(keyStore = keyStore, keyAlias = keyAlias);
+    return crypto:decryptRsaEcb(input, pk, padding = padding);
+}
+
+function testEncryptRsaEcbWithInvalidKey(byte[] input, crypto:RsaPadding padding = "PKCS1", byte[]? iv = ())
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
function testEncryptRsaEcbWithInvalidKey(byte[] input, crypto:RsaPadding padding = "PKCS1", byte[]? iv = ()) returns byte[]|error {
```

File: stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal [hunk 32]
```diff
@@ -139,3 +139,81 @@ function testSignRsaMd5WithInvalidKey(byte[] input) returns byte[]|error {
...
+returns byte[]|error {
```
Comment (NatashaWso2) [inline stdlib/crypto/src/test/resources/test-src/crypto/crypto-test.bal]:
```suggestion
```

File: stdlib/crypto/src/main/ballerina/crypto/Module.md (hunk 22 above)
Comment (NatashaWso2) [reply stdlib/crypto/src/main/ballerina/crypto/Module.md]:
```suggestion
```

**Pull request merged**